LB_DSC_19thApril_WKDY_MR_2024

Mohit Chaudhary 7011883015

Q1. Declare a boolean value and store it in a variable.
Check the type and print the id of the same.

In [37]:
boolean_variable = True   #Declaring a boolean value and storing it in a variable. 

print('The type of variable is:',type(boolean_variable))   #Checking the type of variable.
print('The id of variable is:',id(boolean_variable))       #Checking the id of variable

The type of variable is: <class 'bool'>
The id of variable is: 140710509619744


Q2.  Take one boolean value between 0 - 256.
Assign it to two different variables.
Check the id of both the variables. It should come the same. Check why?

In [11]:
boolean_value1 = True            #Taking one boolean value
boolean_value2 = True             #Assigning it to another variable.


print('The id of first variable is:',id(boolean_value1))   #Checking the id of boolean_value1
print('The id of Second variable is:',id(boolean_value1))   #Checking the id of boolean_value2

The id of first variable is: 140710507457056
The id of Second variable is: 140710507457056


As we can see that both first and second variable is having same id because both variables has same value which is stored in memory location '140710507457056', this is called memory reusibility.

Q3. Arithmetic Operations on boolean data, 
Take two different boolean values, 
Store them in two different variables, 
Do below operations on them:-
   Find sum of both values
   Find difference between them
   Find the product of both.
   Find value after dividing first value with second value
   Find the remainder after dividing first value with second value    Find the quotient after dividing first value with second value    Find the result of first value to the power of second value.

In [35]:
a_bool = bool(int(input('Enter any number between 0-256: ')))  #Taking first value from user.
b_bool = bool(int(input('Enter any number between 0-256: ')))  #Taking second value from user.

print('sum of first and second value is', a_bool + b_bool)                         #Getting the sum of both values
print('Difference of first and second value is', a_bool - b_bool)                  #Finding the differce between both values
print('Product of first and second value is', a_bool * b_bool)                     #Finding the product of both values.
print('Division of first with second value is', a_bool / b_bool)                   #dividing first value with second value
print('Quotient after dividing first value with second value', a_bool // b_bool)   #Quotient after dividing first value with second value
print('Remainder after dividing first value with second value', a_bool % b_bool)   #remainder after dividing first value with second value
print('first value to the power of second value is', a_bool ** b_bool)             #first value to the power of second value.


Enter any number between 0-256:  25
Enter any number between 0-256:  10


sum of first and second value is 2
Difference of first and second value is 0
Product of first and second value is 1
Division of first with second value is 1.0
Quotient after dividing first value with second value 1
Remainder after dividing first value with second value 0
first value to the power of second value is 1


Q4.  Comparison Operators on boolean values
Take two different boolean values.
Store them in two different variables.
Do below operations on them:-
Compare these two values with below operator:- Greater than, '>'
less than, '<'
Greater than or equal to, '>='
Less than or equal to, '<='
Observe their output(return type should be boolean)

In [6]:
bool1 = bool(int(input('Enter any number between 0-256: ')))       #Taking first value from user.
bool2 = bool(int(input('Enter any number between 0-256: ')))       #Taking second value from user.

print('Is first value Greater than second value:', bool1 > bool2)               #Greater than, '>'
print('Is first value Less than second value:', bool1 < bool2)                  #less than, '<'
print('Is first value Less than or equal to second value:', bool1 <= bool2)     #Greater than or equal to, '>='
print('Is first value Greater than or equal to second value', bool1 >= bool2)   #Less than or equal to, '<='


Enter any number between 0-256:  300
Enter any number between 0-256:  123


Is first value Greater than second value: False
Is first value Less than second value: False
Is first value Less than or equal to second value: True
Is first value Greater than or equal to second value True


As we can observe that the output we have obtained in all scenarios is Boolean (True/False), this is because False=0 while rest all values are stored in boolean as True

Q5. Equality Operator
Take two different boolean values.
Store them in two different variables.
Equate them using equality operators (==, !=) Observe the output(return type should be boolean)

In [14]:
value1 = bool(int(input('Enter any number between 0-256: ')))              #Taking first value from user.
value2 = bool(int(input('Enter any number between 0-256: ')))              #Taking second value from user.
print('Checking if values are equal using == :,', value1 == value2)        #Checking if values are equal
print('Checking if values are different using != :', value1 != value2)     #Checking if values are different


Enter any number between 0-256:  45
Enter any number between 0-256:  100


Checking if values are equal using == :, True
Checking if values are different using != : False


As we can observe that the output we have obtained in both scenarios is Boolean (True/False), this is because False=0 while rest all values are stored in boolean as True

Q6. Logical operators
Observe the output of below code
Cross check the output manually
print(True and True)
#----------------------------------------->Output is Trueprint(False and True)
#----------------------------------------->Output is False
print(True and False)
#----------------------------------------->Output is False
print(False and False)
#----------------------------------------->Output is False
print(True or True)
#----------------------------------------->Output is True
print(False or True)
#----------------------------------------->Output is True
print(True or False)
#----------------------------------------->Output is True
print(False or False)
#----------------------------------------->Output is False
print(not True)
#----------------------------------------->Output is False
print(not False)
#----------------------------------------->Output is True

In [27]:
#Lets check the output manually:

print(True and True)        
print(False and True)        
print(True and False)        
print(False and False)       

print(True or True)          
print(False or True)         
print(True or False)         
print(False or False)        

print(not True)              
print(not False)             

True
False
False
False
True
True
True
False
False
True


Q7. Bitwise Operators
Do below operations on the values provided below:-
   Bitwise and(&) --------------> True, True    -------> Output is True    Bitwise or(|)  --------------> True, False   -------> Output is True    Bitwise(^)     --------------> True, False   -------> Output is True    Bitwise negation(~) ---------> True          -------> Output is -2    Bitwise left shift  ---------> True,2        -------> Output is 4    Bitwise right shift ---------> True,2        -------> Output is 0 Cross check the output manually

In [30]:
#Lets check the output manually:

print(True & True)
print(True | False)
print(True ^ False)
print(~True)
print(True << 2)
print(True >> 2)

True
True
True
-2
4
0


Q8. What is the output of expression inside the print statement. Cross check before running the program.
a = True
b = True
print(a is b)          #True or False?   #print(a is not b)      #True or False?

In [33]:
#lets run the expressions given in the question to us:
a = True
b = True
print(a is b)          
print(a is not b)      

a = False
b = False
print(a is b)          
print(a is not b)      

True
False
True
False


Q9. Membership operation
in, not in are two membership operators and it returns boolean value
print(True in [10,10.20,10+20j,'Python', True]) print(False in (10,10.20,10+20j,'Python', False)) print(True in {1,2,3, True})
print(True in {True:100, False:200, True:300}) print(False in {True:100, False:200, True:300})

In [10]:
#lets run the expressions given in the question to us:

print(True in [10,10.20,10+20j,'Python', True])
print(False in (10,10.20,10+20j,'Python', False))
print(True in {1,2,3, True})
print(True in {True:100, False:200, True:300})
print(False in {True:100, False:200, True:300})

True
True
True
True
True
